### ISSUE TITLE
question: how to setup basic authentication

### ISSUE THREAD

**Issue by mouzzampk2014** [greenpau/caddy-security#52]
> A clear and concise description of what you want to accomplish.

Hi, i have few services which I need to put behind authentication. 

Could you please provide a snippet how can I add the basic authentication to one of my service. 

Thank you

**Comment by greenpau** [comment]
@mouzzampk2014 , please check this [video](https://youtu.be/Mxbjfv47YiQ?t=2605) and review this snippet: https://github.com/authp/authp.github.io/blob/main/assets/conf/local/Caddyfile

First, you need to setup local authentication. See this [video](https://www.youtube.com/watch?v=k8tbbffMGZk)

Second, add the `with basic auth` directive in your authorization policy:

```
authorization policy mypolicy {
  with basic auth realm local
}
```

**Comment by mouzzampk2014** [comment]
Hi @greenpau thank you so much for replying to my request.

I looked at the video but little confused that I need the whole section in my Caddyfile or just a block.

Basically thats my Caddy file and I am not sure where I need to add the basic authentication. I would really appreciate if you can guide me with this, thank you :)

```
{
  http_port  8080
  https_port 8443
}

:8080 {
  log {
    output file /config/logs/access.log
  }
  root * /app/www
  file_server
}

overseerr.mouzzampk.xyz {
  log {
    output file /config/logs/access.log
    level WARN
  }
  tls {
	dns cloudflare {API}
  }
  reverse_proxy 192.168.0.43:5055
}
```

**Comment by greenpau** [comment]
@mouzzampk2014 , you would need to get more familiar with caddy and consider using this: https://caddyserver.com/docs/caddyfile/directives/basicauth

**Comment by mouzzampk2014** [comment]
Thanks @greenpau for the link and guidance. I am basically new to caddy so trying to find my way. 

Let me try again and I hope this time I will be okay.

Thanks again

**Comment by greenpau** [comment]
> I am basically new to caddy so trying to find my way.

@mouzzampk2014 , that's what I figured out. You need to get comfortable with build-in plugins, e.g. `respond`, `reverse_proxy`, etc. I recommend asking questions here: https://caddy.community/
